edgexfoundry/edgex-compose
# feat: Add nat-bus option to generate compose files for NATS

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  https://github.com/edgexfoundry/edgex-docs/pull/846


## Testing Instructions
run `make build` from Compose Builder using this branch
Verify the none of the generated compose files changed.
Build ASC, edge-go services, Device virtual docker images with NATS. (add `-tags "include_nats_messaging` to each build target)
run `make gen nats-bus no-secty dev app-dev`
Modify compose file to use dev image of device-virtual
Verify the generated compose file has the proper overrides for NATS
Run `make up` 
Verify all service are connected to NAT Sever w/o error
Change app-rules-engine loglevel to DEBUG
Verify event are flowing
_

> Note eKuiper will not have NATS support yet and will fail to connect

_
